ddsrouter_core/src/cpp/dynamic/DiscoveryDatabase.hpp

```diff
@@ -68,6 +68,13 @@ class DiscoveryDatabase
      */
     ~DiscoveryDatabase();
 
+    /**
+     * @brief Initialize the queue processing thread routine
+     *
+     * Create a thread in charge of processing the dynamic discovery of endpoints.
+     */
+    void init() noexcept;
```

Comment:
I do not really like this. We decided we would not have init methods in classes, so add one "randomly" seems to me like a problem. 
Do one of these:
1. use `enable / disable` methods as the other classes and implement them.
2. comment extensly that this class requires to be init after constructed and why